tests/test_aqmon.py

```diff
@@ -0,0 +1,267 @@
+            # There is one ON without an OFF at the end of the run in
+            # the last chunk. This should fill all the way to the end as
+            # deadtime:
+            dt = self._last_endtime - res['time'][-1]
+            self.TOTAL_DEADTIME += [dt]
+        return self.chunk(start=previous_end,
+                          end=self._last_endtime,
+                          data=res)
+
+
+class DummyEventBasics(strax.Plugin):
+    """Get evenly spaced random duration events that don't overlap"""
+    n_chunks = strax.Config(default=3)
+    event_time_range = strax.Config(default=(0, 3e6),
+                                    help='Where to span the durations of chunks over'
+                                    )
+    events_per_chunk = strax.Config(default=20)
+    event_durations = strax.Config(
+        default=(1000, 20_000),
+        help='event durations (min, max) ns'
+    )
+    depends_on = ()
+    save_when = strax.SaveWhen.ALWAYS
+    provides = 'event_basics'
+    data_kind = 'events'
+    dtype = strax.time_fields
+    _events = None
+
+    def source_finished(self):
+        return True
+
+    def is_ready(self, chunk_i):
+        return chunk_i < self.n_chunks
+
+    def get_events_this_chunk(self, chunk_i):
+        if self._events is None:
+            res = np.zeros(int(self.events_per_chunk * self.n_chunks), dtype=self.dtype)
+            times = np.linspace(*self.event_time_range, len(res))
+            res['time'] = times
+
+            endtimes = res['time'] + np.random.randint(*self.event_durations, size=len(res))
+            # Don't allow overlapping events
+            endtimes[:-1] = np.clip(endtimes[:-1], 0, res['time'][1:] - 1)
+            res['endtime'] = endtimes
+            self._events = np.split(res, self.n_chunks)
+        return self._events[chunk_i]
+
+    def chunk_start(self, chunk_i) -> int:
+        # Get the start time of the requested chunk
+        assert self._events is not None, "run get_events_this_chunk first!"
+        if chunk_i == 0:
+            return self._events[chunk_i]['time'][0]
+        return self._events[chunk_i - 1]['endtime'][-1]
+
+    def get_chunk_end(self, chunk_i):
+        assert self._events is not None, "run get_events_this_chunk first!"
+        if chunk_i == self.n_chunks:
+            return self.event_time_range[1]
+        return self._events[chunk_i]['endtime'][-1]
+
+    def compute(self, chunk_i):
+        events = self.get_events_this_chunk(chunk_i)
+        events_within = events['endtime'] < self.event_time_range[1]
+        events = events[events_within]
+
+        return self.chunk(start=self.chunk_start(chunk_i),
+                          end=self.get_chunk_end(chunk_i),
+                          data=events)
+
+
+class TestAqmonProcessing(TestCase):
+    def setUp(self) -> None:
+        st = straxen.test_utils.nt_test_context().new_context()
+        st.set_context_config({'free_options': list(st.config.keys())})
```

Comment:
Why do you need this?

Reply:
otherwise I get a lot of warnings since I deregister a lot of plugins. I could also have started with a bare context instead of the nT testing one

Reply:
ill comment